### ISSUE TITLE
How to use flash attn in RelPositionMultiHeadedAttention

### ISSUE THREAD

**Issue by hexisyztem** [FunAudioLLM/CosyVoice#181]
As we all know, flash attn calculates: (softmax(Q @ K^T * softmax_scale) @ V), but RelPositionMultiHeadedAttention replaces Q @ K^T with (Q + pos_bias_u) @ K^T + (Q + pos_bias_v) @ P^T.
So now, how can I use flash attn to optimize RelPositionMultiHeadedAttention?

**Comment by hexisyztem** [comment]
By the way, is there a cosyvoice base model that uses multihead attention?

**Comment by hexisyztem** [comment]
I raised an issue in the flash attn project to inquire about this, and the official response was that it is not supported. After reviewing the specific code logic, I confirmed that supporting this part in flash attn would indeed require a significant amount of work.
https://github.com/Dao-AILab/flash-attention/issues/1080

**Comment by aluminumbox** [comment]
torch already support flash attention in version 2.2, check https://pytorch.org/blog/pytorch2-2/ we will also support it later
